### PR TITLE
Copy trivia to rewrite tuples correctly

### PR DESCRIPTION
**Pull request by jjonescz**
Fixes part 1 of https://github.com/dotnet/razor-compiler/issues/315#issuecomment-1233499131 exactly as suggested there.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/razor-compiler?pullRequest=342) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/razor-compiler?pullRequest=342) before we can accept your contribution.<br/><br/>:x: jjonescz [sign now](https://cla.dotnetfoundation.org/dotnet/razor-compiler?pullRequest=342)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/razor-compiler?pullRequest=342) it.</sub>

**Review by 333fred — APPROVED** [review]
Welcome to the team! @dotnet/razor-compiler for a second review.

File: src/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs [hunk 1]
```diff
@@ -379,6 +379,51 @@ @using Test
...
+
+@code {
+    [Parameter]
+    public TParam InferParam { get; set; }
+
+    [Parameter]
+    public RenderFragment<(MyClass I1, MyStruct I2, TParam P)> Template { get; set; }
+}
+
+<TestComponent InferParam=""1"">
```
Comment (RikkiGibson) [inline src/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs:415]: where does the name TestComponent come from? Is it baked into the test helpers?

**Review by RikkiGibson — APPROVED** [review]

File: src/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs (hunk 1 above)
Comment (333fred) [reply src/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs:415]: Yes.

**Pull request merged**